crates/bevy_ecs/src/schedule/stage.rs

```diff
@@ -392,33 +334,34 @@ impl SystemStage {
             })
             .collect();
 
-        for index in self.uninitialized_at_start.drain(..) {
-            let container = &mut self.exclusive_at_start[index];
-            if let Some(index) = container.run_criteria() {
-                container.set_run_criteria(new_indices[index]);
-            }
-            container.system_mut().initialize(world);
-        }
-        for index in self.uninitialized_before_commands.drain(..) {
-            let container = &mut self.exclusive_before_commands[index];
-            if let Some(index) = container.run_criteria() {
-                container.set_run_criteria(new_indices[index]);
-            }
+        for (mut container, system_type) in self.uninitialized_systems.drain(..) {
             container.system_mut().initialize(world);
-        }
-        for index in self.uninitialized_at_end.drain(..) {
-            let container = &mut self.exclusive_at_end[index];
+
             if let Some(index) = container.run_criteria() {
                 container.set_run_criteria(new_indices[index]);
             }
-            container.system_mut().initialize(world);
-        }
-        for index in self.uninitialized_parallel.drain(..) {
-            let container = &mut self.parallel[index];
-            if let Some(index) = container.run_criteria() {
-                container.set_run_criteria(new_indices[index]);
+
+            match system_type {
+                SystemType::Parallel => {
+                    if container.system().is_exclusive() {
+                        // we don't know if exclusive until params are initialized (unless told)
+                        self.exclusive_at_start.push(container);
```

Comment:
This isn't a great default: it differs from the standard position that exclusive systems are inserted.